Sources/SwiftLexicalLookup/IntroducingToSequentialParentScopeSyntax.swift

```diff
@@ -0,0 +1,24 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+protocol IntroducingToSequentialParentScopeSyntax: ScopeSyntax {
+  /// Returns names matching lookup that should be
+  /// handled by it's parent sequential scope.
+  func introducesToSequentialParent(
```

Comment:
I find it a little surprising that `ScopeSyntax` has an API that returns all names introduce in its scope (`introducedNames`) while this API contains a filter. I think it would be more consistent if this also returned all the names introduced to the parent scope and then be named `namesIntroducedToParentScope`. As far as I can tell it wouldn’t need to take any parameters at all then and could be a computed property as well.

Reply:
I think that was poor naming on my part as `introducesToSequentialParent` worked more like `lookup` producing `[LookupResult]` to be interleaved with results of the sequential parent, rather than `introducedNames` that just return all `LookupName`s available at that scope. I renamed `introducesToSequentialParent` to less misleading `lookupFromSequentialParent` and added `namesIntroducedToSequentialParent` that closely resembles `introducedNames` of a normal scope.